workspace_test.go

```diff
@@ -622,3 +622,53 @@ func TestNewWorkspaceConfig(t *testing.T) {
 		assert.Equal(t, output.Valid, true, output.Diagnostics)
 	})
 }
+
+func planWithWorkspaceAction(actions tfjson.Actions) *tfjson.Plan {
+	return &tfjson.Plan{
```

Comment:
Starts to get a bit brittle to be creating these structs manually. You really want to test your `PlanForDeletion` func which is provider agnostic. Would be good to generate these plans from actual HCL Terraform config, somewhat akin to a provider acceptance test. The latency of running the TF CLI seems acceptable. But if not, you could also use a `go:generate` comment to invoke `terraform` and write plan JSON. Then you'd invoke `go generate` to make your fixtures and then `go test` wouldn't have to run the CLI every time.

Reply:
yeah, this is a good push. I think running the tfcli should be fine. Each validate test adds 2ish seconds to the suite so it wont be too bad